modal/functions.py

```diff
@@ -276,7 +332,158 @@ class _Function(_Object, type_prefix="fu"):
     _tag: str
     _raw_f: Callable[..., Any]
     _build_args: dict
-    _parent: "_Function"
+    _can_use_base_function: bool = False
+
+    # when this is the method of a class/object function, invocation of this function
+    # should be using another function id and supply the method name in the FunctionInput:
+    _use_function_id: str  # The function to invoke
+    _use_method_name: str = ""
+
+    # TODO (elias): remove _parent. In case of instance functions, and methods bound on those,
+    #  this references the parent class-function and is used to infer the client for lazy-loaded methods
+    _parent: Optional["_Function"] = None
+
+    def _bind_method(
+        self,
+        user_cls,
+        method_name: str,
+        partial_function: "modal.partial_function._PartialFunction",
+    ):
+        """mdmd:hidden
+
+        Creates a function placeholder function that binds a specific method name to
+        this function for use when invoking the function.
+
+        Should only be used on "class service functions". For "instance service functions",
+        we don't create an actual backend function, and instead do client-side "fake-hydration"
+        only, see _bind_instance_method
+        """
+        assert self._info  # has to be a local function to be able to "bind" it
+        serialized = self._info.is_serialized()
+        full_name = f"{user_cls.__name__}.{method_name}"
+
+        if partial_function.is_generator:
+            function_type = api_pb2.Function.FUNCTION_TYPE_GENERATOR
+        else:
+            function_type = api_pb2.Function.FUNCTION_TYPE_FUNCTION
+
+        class_function = self
+
+        async def _load(method_bound_function: "_Function", resolver: Resolver, existing_object_id: Optional[str]):
+            function_definition = api_pb2.Function(
```

Comment:
Backend comment: Once web labels aren't tied to functions, we could probably replace this load method with something simpler that only creates web labels. Right now it creates an almost empty Function on the backend that only has a _use_function_id and _use_method_name + web label